### PR TITLE
feat: support circom

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR enables circom frontend support.

### CODE REVIEW

File: vendors/circom/circomlib/r1cs/r1cs.h [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+  kWire2LabelIdMap = 0x3,
+  kCustomGatesList = 0x4,
+  kCustomGatesApplication = 0x5,
+};
+
+std::string_view SectionTypeToString(SectionType type);
+
+struct HeaderSection {
+  PrimeField modulus;
+  // Total Number of wires including ONE signal (Index 0).
```
Comment (ashjeong) [inline vendors/circom/circomlib/r1cs/r1cs.h:75]:
```suggestion
  // Total number of wires including ONE signal (Index 0).
  ```
  & all others

File: vendors/circom/circomlib/r1cs/r1cs_parser.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+#include <memory>
+
+#include "circomlib/r1cs/r1cs.h"
+#include "tachyon/base/files/file_path.h"
+
+namespace tachyon::circom {
+
+class R1CSParser {
+ public:
+  // Return nullptr if the parser is failed to parse.
```
Comment (ashjeong) [inline vendors/circom/circomlib/r1cs/r1cs_parser.h:13]:
```suggestion
  // Return nullptr if the parser fails to parse.

**Review by ashjeong — APPROVED** [review]
LGTM

File: vendors/circom/circomlib/r1cs/r1cs.h [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+  kCustomGatesApplication = 0x5,
+};
+
+std::string_view SectionTypeToString(SectionType type);
+
+struct HeaderSection {
+  PrimeField modulus;
+  // Total number of wires including ONE signal (Index 0).
+  uint32_t num_wires;
+  // Total number of wires public output wires. They should be starting at
```
Comment (Insun35) [inline vendors/circom/circomlib/r1cs/r1cs.h:77]: Please remove duplicate words.
```suggestion
  // Total number of public output wires. They should be starting at
```

File: vendors/circom/circomlib/r1cs/r1cs.h [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+  // idx 1.
+  uint32_t num_public_outputs;
+  // Total number of wires public input wires. They should be starting just
```
Comment (Insun35) [inline vendors/circom/circomlib/r1cs/r1cs.h:80]: Please remove duplicate words.
```suggestion
  // Total number of public input wires. They should be starting just
```

File: vendors/circom/circomlib/r1cs/r1cs.h [hunk 5]
```diff
@@ -0,0 +1,363 @@
...
+  // after the public output.
+  uint32_t num_public_inputs;
+  // Total number of wires private input wires. They should be starting just
```
Comment (Insun35) [inline vendors/circom/circomlib/r1cs/r1cs.h:83]: Please remove duplicate words.
```suggestion
  // Total number of private input wires. They should be starting just
```

**Review by Insun35 — CHANGES_REQUESTED** [review]

**Review by Insun35 — APPROVED** [review]
LGTM

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Pull request merged**